code/modules/mob/living/carbon/damage_procs.dm

```diff
@@ -111,90 +133,79 @@
 //Heals ONE bodypart randomly selected from damaged ones.
 //It automatically updates damage overlays if necessary
 //It automatically updates health status
-/mob/living/carbon/heal_bodypart_damage(brute, burn, only_robotic = 0, only_organic = 1)
+/mob/living/carbon/heal_bodypart_damage(brute = 0, burn = 0, stamina = 0, updating_health = TRUE, only_robotic = FALSE, only_organic = TRUE)
 	var/list/obj/item/bodypart/parts = get_damaged_bodyparts(brute,burn)
 	if(!parts.len)
 		return
 	var/obj/item/bodypart/picked = pick(parts)
-	if(picked.heal_damage(brute, burn, only_robotic, only_organic))
+	if(picked.heal_damage(brute, burn, stamina, only_robotic, only_organic))
 		update_damage_overlays()
 
 //Damages ONE bodypart randomly selected from damagable ones.
 //It automatically updates damage overlays if necessary
 //It automatically updates health status
-/mob/living/carbon/take_bodypart_damage(brute, burn)
+/mob/living/carbon/take_bodypart_damage(brute = 0, burn = 0, stamina = 0)
 	var/list/obj/item/bodypart/parts = get_damageable_bodyparts()
 	if(!parts.len)
 		return
 	var/obj/item/bodypart/picked = pick(parts)
-	if(picked.receive_damage(brute,burn))
+	if(picked.receive_damage(brute, burn, stamina))
 		update_damage_overlays()
 
-
 //Heal MANY bodyparts, in random order
-/mob/living/carbon/heal_overall_damage(brute, burn, only_robotic = 0, only_organic = 1, updating_health = 1)
-	var/list/obj/item/bodypart/parts = get_damaged_bodyparts(brute,burn)
+/mob/living/carbon/heal_overall_damage(brute = 0, burn = 0, stamina = 0, only_robotic = FALSE, only_organic = TRUE, updating_health = TRUE)
+	var/list/obj/item/bodypart/parts = get_damaged_bodyparts(brute, burn, stamina)
 
-	var/update = 0
-	while(parts.len && (brute>0 || burn>0) )
+	var/update = NONE
+	while(parts.len && (brute > 0 || burn > 0 || stamina > 0))
 		var/obj/item/bodypart/picked = pick(parts)
 
 		var/brute_was = picked.brute_dam
 		var/burn_was = picked.burn_dam
+		var/stamina_was = picked.stamina_dam
 
-		update |= picked.heal_damage(brute,burn, only_robotic, only_organic, 0)
+		update |= picked.heal_damage(brute, burn, stamina, only_robotic, only_organic, FALSE)
 
-		brute -= (brute_was-picked.brute_dam)
-		burn -= (burn_was-picked.burn_dam)
+		brute -= (brute_was - picked.brute_dam)
+		burn -= (burn_was - picked.burn_dam)
+		stamina -= (stamina_was - picked.stamina_dam)
 
 		parts -= picked
 	if(updating_health)
 		updatehealth()
+		update_stamina()
```

Comment:
update_stamina should *ALWAYS* be called after a health update, otherwise full heals on someone in staminacrit will result in them being permanently stuck in stamina crit, among many other issues. Remove one tab worth of whitespace from this proc and add `//CIT CHANGE - makes sure update_stamina() always gets called after a health update`